src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2361 @@
+        self_attn (bool, *optional*): Enable Tiny self attention in addition to MLP mixing. Defaults to False.
+        self_attn_heads (bool, *optional*): Self attention heads. Defaults to 1.
+        norm_mlp (str, *optional*): Norm layer (BatchNorm or LayerNorm). Defaults to LayerNorm.
+    """
+
+    # @get_class_params
+    def __init__(
+        self,
+        num_patches: int,
+        patch_size: int = 16,
+        in_channels: int = 3,
+        num_features: int = 128,
+        expansion_factor: int = 2,
+        num_layers: int = 8,
+        dropout: float = 0.5,
+        mode: str = "common_channel",
+        gated_attn: bool = False,
+        self_attn: bool = False,
+        self_attn_heads: int = 1,
+        norm_mlp="LayerNorm",
+        use_pe: bool = False,
+        pe: str = "zeros",
+        learn_pe: bool = False,
+    ):
+        super().__init__()
+
+        ffn = "mlp"
+        mixer_type = "base"
+
+        # if mode == "flatten":
+        #     logger.warn("Use mode = common_channel or mix_channel. mode=flatten is not preferred due to poor performance")
+
+        self.mode = mode
+        self.use_pe = use_pe
+
+        if mode == "flatten":
+            self.patcher = nn.Linear(in_channels * patch_size, num_features)
+
+        elif mode in ["common_channel", "mix_channel"]:
+            self.patcher = nn.Linear(patch_size, num_features)
+
+        self.num_patches = num_patches
+        self.patch_size = patch_size
+        self.in_channels = in_channels
+        self.num_features = num_features
+        self.num_layers = num_layers
+
+        self.mlp_mixer_encoder = PatchTSMixerBackbone(
+            num_patches=num_patches,
+            patch_size=patch_size,
+            in_channels=in_channels,
+            num_features=num_features,
+            expansion_factor=expansion_factor,
+            num_layers=num_layers,
+            dropout=dropout,
+            mode=mode,
+            gated_attn=gated_attn,
+            ffn=ffn,
+            self_attn=self_attn,
+            self_attn_heads=self_attn_heads,
+            mixer_type=mixer_type,
+            norm_mlp=norm_mlp,
+        )
+
+        if use_pe:
+            self.W_pos = positional_encoding(pe, learn_pe, num_patches, num_features)
+
+    def forward(self, x, output_hidden_states: Optional[bool] = False):
+        # x: [bs  x n_vars x num_patch x patch_len]
+        batch_size = x.shape[0]
+        logger.debug(x.shape)
+
+        if self.mode == "flatten":
+            x = x.permute(0, 2, 1, 3)  # x: [bs  x num_patch x n_vars  x patch_len]
+            x = torch.reshape(
+                x, (batch_size, self.num_patches, self.in_channels * self.patch_size)
+            )  # x: [bs x num_patch x patch_len * n_vars]
+
+        # elif self.mode in ["common_channel", "mix_channel"]:
+        #     x = x.permute(0, 2, 1, 3)  # x: [bs x n_vars x num_patch x patch_len]
+
+        logger.debug(x.shape)
```

Comment:
can be removed

Reply:
https://github.com/huggingface/transformers/pull/26247/commits/46360d694e1e7f4464b268251bc8b6132db00461